File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/com/automatetheplanet/_1_page_object_pattern/Pages/BingMainPage/BingMainPageElementMap.java
package com.automatetheplanet._1_page_object_pattern.Pages.BingMainPage;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class BingMainPageElementMap {

    private final WebDriver _browser;

    public BingMainPageElementMap(WebDriver browser) {
        _browser = browser;
    }

    public WebElement searchBox() {
        return _browser.findElement(By.id("sb_form_q"));
    }

    public WebElement goButton() {
        return _browser.findElement(By.xpath("//label[@for='sb_form_go']"));
    }

    public WebElement resultsCountDiv() {
        return _browser.findElement(By.id("b_tween"));
    }

}

File: dotnet/DesignPatternsInAutomatedTesting-Series/SingletonDesignPattern/Pages/BingMainPage/BingMainPageElementMap.cs
﻿// <copyright file="BingMainPageElementMap.cs" company="Automate The Planet Ltd.">
// Copyright 2020 Automate The Planet Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// You may not use this file except in compliance with the License.
// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// </copyright>
// <author>Anton Angelov</author>
// <site>http://automatetheplanet.com/</site>

using OpenQA.Selenium;
using SingletonDesignPattern.Core;

namespace SingletonDesignPattern.Pages.BingMainPage
{
    public class BingMainPageElementMap : BasePageElementMap
    {
        public IWebElement SearchBox 
        {
            get
            {
                return Browser.FindElement(By.Id("sb_form_q"));
            }
        }

        public IWebElement GoButton 
        {
            get
            {
                return Browser.FindElement(By.Id("sb_form_go"));
            }
        }
       
        public IWebElement ResultsCountDiv
        {
            get
            {
                return Browser.FindElement(By.Id("b_tween"));
            }
        }
    }
}